### PR TITLE
Add libpqxx/7.4.0

### PR DESCRIPTION
**Pull request by tt4g**
Specify library name and version:  **libpqxx/7.4.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by mathbunnyru — DISMISSED** [review]

File: recipes/libpqxx/all/patches/fix-apple-clang-compilation-7.4.0.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff --git a/include/pqxx/connection.hxx b/include/pqxx/connection.hxx
+index b44daa18..21dc1fd3 100644
+--- a/include/pqxx/connection.hxx
++++ b/include/pqxx/connection.hxx
+@@ -33,6 +33,7 @@
+ #include "pqxx/errorhandler.hxx"
+ #include "pqxx/except.hxx"
+ #include "pqxx/prepared_statement.hxx"
++#include "pqxx/separated_list.hxx"
```
Comment (mathbunnyru) [inline recipes/libpqxx/all/patches/fix-apple-clang-compilation-7.4.0.patch:9]: Could you also send a PR in the upstream, please?

**Review by mathbunnyru — DISMISSED** [review]

File: recipes/libpqxx/all/patches/fix-apple-clang-compilation-7.4.0.patch (hunk 1 above)
Comment (tt4g) [reply recipes/libpqxx/all/patches/fix-apple-clang-compilation-7.4.0.patch:9]: Yes. I'm going to check the results of the CI build before PR.

**Comment by tt4g** [comment]
It is strange that `std::numeric_limits` is not found.
Was it not a good idea to make it a draft?

File: recipes/libpqxx/all/patches/fix-apple-clang-compilation-7.4.0.patch (hunk 1 above)
Comment (tt4g) [reply recipes/libpqxx/all/patches/fix-apple-clang-compilation-7.4.0.patch:10]: This was fixed before PR: jtv/libpqxx@cbd2b35ef5bce8a1fc8bfd1efd804d965976ce23

**Comment by tt4g** [comment]
It may be related to the failure of the CI build: https://github.com/conan-io/conan-center-index/pull/4401#issuecomment-784218451

@prince-chrismc Is it a CI problem that `std::numeric_limits` is not found?

**Comment by prince-chrismc** [comment]
https://github.com/jtv/libpqxx/blob/50283a60cd55d7d6bb1779ccee58d20c5df2f244/include/pqxx/util.hxx#L21

Seems like it =?

looks like a new compiler...

You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

**Comment by mathbunnyru** [comment]
> It is strange that `std::numeric_limits` is not found.
> Was it not a good idea to make it a draft?

Actually, the fix is quite easy.
`std::numeric_limits` is located in `<limits>` header file (yes, this is odd, C++ ❤️ ).

And I think some old versions did not include this header, so you need to create a small patch/replacement to include this file.

**Comment by mathbunnyru** [comment]
https://github.com/jtv/libpqxx/commit/143e139fafadfb153a5d0c84c429e6ced3ed06fc

So, needed include was only added in 7.0.5

**Comment by tt4g** [comment]
> So, needed include was only added in 7.0.5

Thank you for your help. I was not aware of this problem.
I will try to add the patch.

**Review by mathbunnyru — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**